File: PetCafeProject/Models/Adocao.cs
using System.ComponentModel.DataAnnotations;

namespace PetCafeProject.Models
{
    public class Adocao
    {
        [Key]
        public int? Codigo { get; set; }
        public string? ClienteCPF{get; set;} 
        public Cliente? Cliente { get; set; }
        public int? AnimalID{get;set;}
        public Animal? Animal { get; set; }
        public string? Data { get; set; }
        public Adocao() { }
    
        public Adocao(Cliente cliente, Animal animal, string data)
        { 
            ClienteCPF = ClienteCPF;
            AnimalID = AnimalID;
            Cliente = cliente;
            Animal = animal;
            Data = data;
        }
    }
}

File: AppPetCafe/src/app/Model/Adocao.ts
import { Cliente } from './Cliente';
import { Animal } from './Animal';

export class Adocao {
  codigo: number = 0;
  clienteCPF: string = '';
  cliente: Cliente | undefined;
  animalID: number = 0;
  animal: Animal | undefined;
  data: string = '';
}
